### PR TITLE
EOS-21188 : Populate motr client instances in CDF based on ConfStore key

### PR DESCRIPTION
**Pull request by supriyachavan4398**
Solution : 'cortx>software>motr>service>client_instances' should be used to populate number of motr client instances in CDF.
In this task we will read above value and use it while generating CDF in 'config' stage of hare mini-provisioning.

Signed-off-by: Supriya Yadav <supriya.s.chavan@seagate.com>

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: provisioning/miniprov/hare_mp/cdf.py [hunk 1]
```diff
@@ -291,4 +291,6 @@ def _create_node(self, machine_id: str) -> NodeDesc:
             # TODO in the future the value must be taken from a correct
             # ConfStore key (it doesn't exist now).
             s3_instances=int(
-                store.get(f'server_node>{machine_id}>s3_instances')))
+                store.get(f'server_node>{machine_id}>s3_instances')),
+            m0_clients=int(
+                store.get(f'server_node>{machine_id}>m0_clients')))
```
Comment (knekrasov) [inline provisioning/miniprov/hare_mp/cdf.py:296]: @supriyachavan4398 Does this key exist in ConfStore nowadays?

Comment (supriyachavan4398) [reply provisioning/miniprov/hare_mp/cdf.py:296]: No, Manually I added this for testing.

Comment (knekrasov) [reply provisioning/miniprov/hare_mp/cdf.py:296]: Yes, I saw your code. I meant a bit different thing.

The list of keys that ConfStore can potentially have is confirmed by other people. As far as I understand the latest "snapshot" of all confirmed keys can be seen here https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/85000402/Mini-Provisioner+ConfStore+Keys
And I don't see `m0_clients` there. Which means that we'll need to initiate the corresponding discussions...

Comment (supriyachavan4398) [reply provisioning/miniprov/hare_mp/cdf.py:296]: Okay Sure, I'll look into it.

**Review by knekrasov — APPROVED** [review]

File: provisioning/miniprov/hare_mp/cdf.py (hunk 1 above)
Comment (SwapnilGaonkar7) [reply provisioning/miniprov/hare_mp/cdf.py:296]: @supriyachavan4398 We can modify our code such that if key is not present in confstore then we will use default hardcoded value.

Comment (knekrasov) [reply provisioning/miniprov/hare_mp/cdf.py:296]: @supriyachavan4398 both [EOS-21188](https://jts.seagate.com/browse/EOS-21188) and [the doc](https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/85000402/Mini-Provisioner+ConfStore+Keys) mention `cortx>software>motr>service>client_instances`.
Why do you refer to `m0_clients`?

Comment (supriyachavan4398) [reply provisioning/miniprov/hare_mp/cdf.py]: I addressed this comment and m0_clients replaced with client_instances.

File: provisioning/miniprov/hare_mp/cdf.py [hunk 2]
```diff
@@ -258,6 +258,12 @@ def _create_node(self, machine_id: str) -> NodeDesc:
         hostname = store.get(f'server_node>{machine_id}>hostname')
         name = store.get(f'server_node>{machine_id}>name')
         iface = self._get_iface(machine_id)
+        try:
+            no_m0clients = int(store.get(
+                f'server_node>{machine_id}>client_instances',
```
Comment (mssawant) [inline provisioning/miniprov/hare_mp/cdf.py:263]: @supriyachavan4398 , conf-store key `cortx>software>motr>service>client_instances` is different in https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/85000402/Mini-Provisioner+ConfStore+Keys and the change is being tracked by https://jts.seagate.com/browse/EOS-21908, which is not compete yet. Then why are we addressing this issue now?
Even the old key given in the ticket is different than what we are using here. I think atleast change this one to the old key as mentioned in the confluence page or differ this change until EOS-21908 is fixed. cc @vaibhavparatwar

Comment (supriyachavan4398) [reply provisioning/miniprov/hare_mp/cdf.py]: @mssawant , I looked at your comment and made changes accordingly. Now I used new Key `cortx>software>motr>service>client_instances` for that.

**Comment by mssawant** [comment]
@supriyachavan4398,
kindly reformat the commit message as follows,
```
Solution : 
Refer to Conf store KV 'cortx>software>motr>service>client_instances' to populate number of
motr client instances while generating CDF in hare mini-provisioner's `config` phase.
```
Please maintain 80 columns restriction.

**Review by mssawant — APPROVED** [review]
@supriyachavan4398, Patch looks good but we need to make sure that the changes are in sync with the provisioner too so that main deployment does not break. Somehow the the VM deployment from the PR doesn't seem to be working. Try rebasing and pushing the branch so that vm deployment is re-triggered or we need to try out a deployment from custom branch using this patch.

**Comment by mssawant** [comment]
@supriyachavan4398 , kindly fixup the commits and resolve the conflicts.

**Comment by vaibhavparatwar** [comment]
image
Deployment successful with Build#2062

**Pull request merged**